noobaa/noobaa-core
# Update Ceph S3 Version to Newest

### Explain the changes
1. update CEPH_TESTS_VERSION to the latest version
2. update the testing script to use the new testing framework tox (see https://github.com/ceph/s3-tests#readme for more details)
3. changed test names to the new format: use s3tests_boto3/functional/test_s3.py::test_basic_key_count instead of s3tests_boto3.functional.test_s3:test_basic_key_count
4. removed S3_CEPH_TEST_STEMS as changing test names is no longer needed. moved tests that previously didn't run because they didn't match its regex (but showed as if they did) to the blacklist. those are the tests related to iam and sts
5. added new tests that don't pass to the pending list
6. changed s3-tests repo permissions, because write permission is required for tox to create files
7. changed manually changing of test_s3select.py in test_ceph_s3_deploy.sh from line 14 to line 16 because the file has been changed
8. changed keyword used to identify skipped tests from 'SKIP' to 'skipped' in order to identify skipped tests 
9. moved test_count_json_operation and test_column_sum_min_max tests to the pending list, since they no longer pass
10. change the way we only run boto3 tests as described in https://github.com/ceph/s3-tests/pull/482/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR44-R48
11. change the check of aws4 to check by test names ending with aws4 instead of using the list. moved aws4 tests to the pending list because of the problem with tox trying to parse the endpoint as if it is an Amazon URL. s3-tests issue: https://github.com/ceph/s3-tests/issues/520 
12. update the commit of the operator to the latest in [ceph-s3-tests.yaml](https://github.com/noobaa/noobaa-core/pull/7326/files#diff-0ac77969a6947e90f1dea7400e179822039effba8c0eef423e3e85e20a27cae6)
13. remove filtering based on not fails_on_rgw on tests job since it is not relevant any more
14. update [ceph_s3_tests_guide.md](https://github.com/noobaa/noobaa-core/compare/master...nadavMiz:noobaa-core:new-ceph-tests?expand=1#diff-84131852147931065e9bcbfcb2778c9447c15ae299a2376a36d079603dd29b08) with latest changes

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1. Run ceph s3 tests job (https://github.com/noobaa/noobaa-core/blob/master/docs/dev_guide/ceph_s3_tests/ceph_s3_tests_guide.md#run-all-ceph-s3-tests)
2. All tests should pass
